drivers/storage/portworx/component/plugin.go

```diff
@@ -0,0 +1,338 @@
+	}
+	if err := k8s.DeleteService(p.client, PluginServiceName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+
+	// delete nginx configs
+	if err := k8s.DeleteDeployment(p.client, NginxDeploymentName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+	if err := k8s.DeleteConfigMap(p.client, NginxConfigMapName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+	if err := k8s.DeleteService(p.client, NginxServiceName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+
+	// delete console plugin
+	if err := k8s.DeleteConsolePlugin(p.client, PluginName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+	if len(errList) > 0 {
+		return commonerrors.New(strings.Join(errList, " , "))
+	}
+	return nil
+}
+
+func (p *plugin) MarkDeleted() {
+	p.isOperatorCreated = false
+}
+
+// RegisterPortworxPluginComponent registers the PortworxPlugin component
+func RegisterPortworxPluginComponent() {
+	Register(PluginComponentName, &plugin{})
+}
+
+func init() {
+	RegisterPortworxPluginComponent()
+}
+
+func (p *plugin) createDeployment(filename, deploymentName string, ownerRef *metav1.OwnerReference, cluster *corev1.StorageCluster) error {
+	deployment, err := k8s.GetDeploymentFromFile(filename, pxutil.SpecsBaseDir())
+	if err != nil {
+		return err
+	}
+	deployment.Namespace = cluster.Namespace
+	deployment.OwnerReferences = []metav1.OwnerReference{*ownerRef}
+
+	existingDeployment := &appsv1.Deployment{}
+	getErr := p.client.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      deploymentName,
+			Namespace: cluster.Namespace,
+		},
+		existingDeployment,
+	)
+	if getErr != nil && !errors.IsNotFound(getErr) {
+		return getErr
+	}
+
+	equal, _ := util.DeepEqualPodTemplate(&deployment.Spec.Template, &existingDeployment.Spec.Template)
+
+	if cluster.Spec.ImagePullSecret != nil && *cluster.Spec.ImagePullSecret != "" {
+		deployment.Spec.Template.Spec.ImagePullSecrets = append(
+			[]v1.LocalObjectReference{},
+			v1.LocalObjectReference{
+				Name: *cluster.Spec.ImagePullSecret,
+			},
+		)
+	}
+
+	if cluster.Spec.Placement != nil {
+		if cluster.Spec.Placement.NodeAffinity != nil {
+			deployment.Spec.Template.Spec.Affinity = &v1.Affinity{
+				NodeAffinity: cluster.Spec.Placement.NodeAffinity.DeepCopy(),
+			}
+		}
+
+		if len(cluster.Spec.Placement.Tolerations) > 0 {
+			deployment.Spec.Template.Spec.Tolerations = make([]v1.Toleration, 0)
+			for _, toleration := range cluster.Spec.Placement.Tolerations {
+				deployment.Spec.Template.Spec.Tolerations = append(
+					deployment.Spec.Template.Spec.Tolerations,
+					*(toleration.DeepCopy()),
+				)
+			}
+		}
+	}
+
+	if !equal {
+		if err := k8s.CreateOrUpdateDeployment(p.client, deployment, ownerRef); err != nil {
+			return err
+		}
+	}
+	p.isOperatorCreated = true
```

Comment:
This flag is not even used anywhere. We should use to avoid creating the Deployment above if already created.

Also, we are using these method for two deployments and using a common flag. We should have a separate flag for each.